fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/ProtocolSupport.java

```diff
@@ -0,0 +1,148 @@
+     *
+     * @param protocolName The name of the protocol to test for.
+     *
+     * @return TRUE if this metric directly or indirectly supports this signal, FALSE if it refuses, MAYBE if it
+     * doesn't assert authority.
+     */
+    public Accepts accepts(String protocolName) {
+        return protocolMap.containsKey(protocolName) ? Accepts.TRUE :
+                blacklist.contains(protocolName) ?
+                        Accepts.FALSE :
+                        Accepts.MAYBE;
+    }
+
+    /**
+     * Create a modified protocol handler which doesn't accepts a certain protocol.
+     *
+     * @param protocolName  The name of a protocol to not handle.
+     *
+     * @return A protocol handler with additional signals bound.
+     */
+    public ProtocolSupport withoutProtocol(String protocolName) {
+        return withoutProtocols(Collections.singleton(protocolName));
+    }
+
+    /**
+     * Create a modified protocol handler which doesn't accepts certain protocols.
+     *
+     * @param protocolNames  The names of protocols to not handle.
+     *
+     * @return A protocol handler with protocols not supported.
+     */
+    public ProtocolSupport withoutProtocols(Collection<String> protocolNames) {
+
+        List<Protocol> protocols =
+                protocolMap.values().stream()
+                        .filter(protocol -> !protocolNames.contains(protocol.getContractName()))
+                        .collect(Collectors.toList());
+        List<String> newBlackList = Stream.concat(protocolNames.stream(), blacklist.stream())
+                .collect(Collectors.toList());
+
+        return new ProtocolSupport(protocols, newBlackList);
+    }
+
+    /**
+     * Create a modified protocol handler which doesn't accepts protocols blacklisted in these other protocols.
+     *
+     * @param protocolSupport  The protocol supports whose blacklists should not be handled.
+     *
+     * @return A protocol handler with additional protocols not supported.
+     */
+    public ProtocolSupport combineBlacklists(Collection<ProtocolSupport> protocolSupport) {
+
+        List<String> protocols =
+                protocolSupport.stream()
+                        .flatMap(support -> support.blacklist.stream())
+                        .collect(Collectors.toList());
+        return withoutProtocols(protocols);
+    }
+
+    /**
+     * Create a modified protocol handler which accepts certain protocols.
+     *
+     * @param addedProtocols  The protocols to handle.
+     *
+     * @return A protocol handler with additional protocols supported.
+     */
+    public ProtocolSupport withProtocols(Collection<Protocol> addedProtocols) {
+
+        Collection<Protocol> newProtocols = Stream.concat(addedProtocols.stream(), protocolMap.values().stream())
+                .collect(Collectors.toSet());
+
+        Collection<String> newBlackList = blacklist.stream()
+                .filter(name -> !addedProtocols.contains(name))
```

Comment:
Shouldn't we be converting `addedProtocols` to a `Collection<String>` first? It looks to me like
we're trying to check that a collection of Protocols contains a String.

Reply:
We're collecting new protocols into with the existing one in line 128 and we're pruning blacklisted names in the second stream.